p/api/fever.php

```diff
@@ -0,0 +1,894 @@
+		$stm->execute();
+		$result = $stm->fetchAll(PDO::FETCH_COLUMN);
+
+		return $result;
+	}
+
+	/**
+	 * @return []
+	 */
+	public function getStarred()
+	{
+		$sql = 'SELECT id FROM `' . $this->prefix . 'entry` WHERE is_favorite=1';
+		$stm = $this->bd->prepare($sql);
+		$stm->execute();
+		$result = $stm->fetchAll(PDO::FETCH_COLUMN);
+
+		return $result;
+	}
+
+	/**
+	 * TODO this is ugly
+	 */
+	protected function bindParamArray($prefix, $values, &$bindArray)
+	{
+		$str = "";
+		foreach ($values as $index => $value) {
+			$str .= ":".$prefix.$index.",";
+			$bindArray[$prefix.$index] = $value;
+		}
+		return rtrim($str, ",");
+	}
+
+	/**
+	 * @param array $feed_ids
+	 * @param array $entry_ids
+	 * @param int|null $max_id
+	 * @param int|null $since_id
+	 * @return FreshRSS_Entry[]
+	 */
+	public function findEntries(array $feed_ids, array $entry_ids, $max_id, $since_id)
+	{
+		$values = [];
+		$order = '';
+
+		$sql = 'SELECT id, guid, title, author, '
+			. ($this->isCompressed() ? 'UNCOMPRESS(content_bin) AS content' : 'content')
+			. ', link, date, is_read, is_favorite, id_feed, tags '
+			. 'FROM `' . $this->prefix . 'entry` WHERE';
+
+		if (!empty($entry_ids)) {
+			$bindEntryIds = $this->bindParamArray("id", $entry_ids, $values);
+			$sql .= " id IN($bindEntryIds)";
+		} else if (!empty($max_id)) {
+			$sql .= ' id < :id';
+			$values[':id'] = $max_id;
+			$order = ' ORDER BY id DESC';
+		} else {
+			$sql .= ' id > :id';
+			$values[':id'] = $since_id;
+			$order = ' ORDER BY id ASC';
+		}
+
+		if (!empty($feed_ids)) {
+			$bindFeedIds = $this->bindParamArray("feed", $feed_ids, $values);
+			$sql .= " AND id_feed IN($bindFeedIds)";
+		}
+
+		$sql .= $order;
+		$sql .= $this->getSelectLimit($max_id, $since_id);
+
+		$stm = $this->bd->prepare($sql);
+		$stm->execute($values);
+		$result = $stm->fetchAll(PDO::FETCH_ASSOC);
+
+		$entries = [];
+		foreach ($result as $dao) {
+			$entries[] = self::daoToEntry($dao);
+		}
+
+		return $entries;
+	}
+
+	/**
+	 * Can be overwritten to support clients that misbehave when using the API.
+	 *
+	 * @param $max_id
+	 * @param $since_id
+	 * @return string
+	 */
+	protected function getSelectLimit($max_id, $since_id)
+	{
+		return ' LIMIT 50';
+	}
+}
+
+/**
+ * Class FeverAPI
+ *
+ * API Password must be the result of the md5 sum of your FreshRSS "username:your-api-password"
+ *
+ * md5 -s "kevin:test-fever"
+ * MD5 ("kevin:test-fever") = 7cd3293ae3c2f3d6648e76f9be65ca59
+ */
+class FeverAPI
+{
+	const API_LEVEL = 3;
+	const STATUS_OK = 1;
+	const STATUS_ERR = 0;
+
+	/**
+	 * whether the API was requested with XML as return value (true) or as JSON (false)
+	 * @var bool
+	 */
+	private $xml = false;
+
+	/**
+	 * FeverAPI constructor executes authentication and initialization.
+	 */
+	public function __construct()
+	{
+		$this->authenticate();
+
+		// are we xml or json?
+		if (isset($_REQUEST["api"]) && strtolower($_REQUEST['api']) === 'xml') {
+			$this->xml = true;
```

Comment:
All the clients I have tested for the Google Reader API only support JSON. It would be nice to check what the clients, which we would like to support, are actually using. If all of them are using e.g. JSON, we could drop the other serialization.

Reply:
None that I tested used XML. I , but support is already baked in: so why remove it?

Reply:
Not very important indeed, but if nobody uses a code, removing it reduces the amount of code to maintain as well as the risks of bugs and other vulnerabilities. If we do not have any client supporting XML, it is not even easy to test that it actually works as intended.

Reply:
I don't mind removing it, it makes things easier.

Reply:
If we cannot find any client using it so yes, I would indeed rather like removing the code